packages/core/src/modules/joblisting/joblisting-location-repository.ts

```diff
@@ -0,0 +1,113 @@
+import { Database } from "@dotkomonline/db"
+import { JobListingId } from "@dotkomonline/types"
+import { Kysely } from "kysely"
+
+export interface JobListingLocationRepository {
+  create(jobListingId: JobListingId, locations: string[]): Promise<void>
+  add(id: JobListingId, locations: string[]): Promise<void>
+  remove(id: JobListingId, locations: string[]): Promise<void>
+  getAll(): Promise<string[]>
+}
+
+export class JobListingLocationRepositoryImpl implements JobListingLocationRepository {
+  constructor(private readonly db: Kysely<Database>) {}
+
+  async getAll(): Promise<string[]> {
+    const locations = await this.db.selectFrom("jobListingLocation").select("name").orderBy("name", "asc").execute()
+    return locations.map((x) => x.name)
+  }
+
+  async create(jobListingId: JobListingId, locations: string[]): Promise<void> {
+    // Start a transaction because you might be dealing with multiple inserts
+    this.db.transaction().execute(async (trx) => {
```

Comment:
needs an await on db.transaction.execute() also create should always return the thing that was created.